src/blockdiagonal.jl

```diff
@@ -102,6 +107,23 @@ end
 
 ## Base
 Base.Matrix(B::BlockDiagonal) = cat(blocks(B)...; dims=(1, 2))
+
+function ChainRulesCore.rrule(::Type{<:Base.Matrix}, B::BlockDiagonal)
+    function Matrix_pullback(Δ::AbstractMatrix)
+        nrows = size.(B.blocks, 1)
+        ncols = size.(B.blocks, 2)
+        row_idxs = cumsum(nrows) .- nrows .+ 1
+        col_idxs = cumsum(ncols) .- ncols .+ 1
```

Comment:
It we wanted to, we could avoid closing over `B` if we moved these outside the pullback.
and replaced `n in eachindex(B.blocks)` with `n in eachindex(row_idxs)`.

Reply:
Very good points. Will make the changes.